documentation/guides/clock_systems.md

```diff
@@ -0,0 +1,275 @@
+config.cpu.clock =
+   sjsu::lpc40xx::SystemController::CpuClockSelect::kSystemClock;
+// Make sure to utilize the internal oscillator only.
+config.system_oscillator =
+   sjsu::lpc40xx::SystemController::OscillatorSource::kIrc;
+// Make sure PLL0 is enabled and available
+config.pll[0].enabled  = true;
+config.pll[0].multiply = 4;
+```
+
+Finally, in order to actually use these new settings, we have two methods:
+
+#### Method 1
+
+```C++
+auto result = system.Initialize();
+if (!result)
+{
+  // Handle error ...
+}
+```
+
+This option seems like the obvious choice but has a fatal issue. During platform
+initialization, peripherals and other drivers that are enabled at startup will
+have adjusted their own clock frequencies based on the frequency they were fed
+initially. If you change these without also rerunning the those driver's
+initialize methods, their clocks will be incorrect and they will not performed
+as expected. The behavior is undefined and can potentially lock up the platform.
+
+#### Method 2 (best practice)
+```C++
+sjsu::InitializePlatform();
+```
+
+Calling the `InitializePlatform()` function directly, will prevent the issues in
+Method 1, since `InitializePlatform()` will handle re-initializing the
+peripherals based on the new clock settings.
+
+!!! Note
+    This only re-initializes the peripherals used at startup. These WILL NOT
+    re-initialize all of the peripherals and driver you have in your
+    application.
+
+## Clock Configuration Examples
+Clock configuration examples can be found in the `demos/` folder with the name
+`clock_configuration`.
+
+- [demos/stm32f10x/clock_configuration](https://github.com/kammce/SJSU-Dev2/tree/master/demos/stm32f10x/clock_configuration)
+- [demos/lpc40xx/clock_configuration](https://github.com/kammce/SJSU-Dev2/tree/master/demos/sjtwo/clock_configuration)
+
+## Best practices
+
+### Best time to change clocks
+The best time to change the clock rate for the platform is at the beginning of
+the main() function before any peripherals or drivers have been initialized.
+Prefer to only change the clock rate once during runtime.
+
+Note that if you change the operating frequency of the system after peripherals
+or other drivers have already been initialized, then those drivers will have
+been setup with a frequency that does not match the actual state of the device.
+In this case, you either need to re-run the initialization of each of the
+drivers and peripherals in your application or simply set the clocks before
+initialization, which tends to reduce code complexity and size. Failure to do
+this will lead to undefined behavior.
+
+### Changing clock rates in a thread
+First off, DON'T! But if you feel the need to, prefer to have just 1 thread
+manage this, or have a mutex lock usage of this resource. The clock tree is a
+single hardware resource and the configuration object is a single statically
+allocated objects, thus are not thread safe to manipulate without some sort of
```

Comment:
allocated **object**, thus are not thread safe to manipulate without some sort of